tests/kernel/arm_irq_vector_table/src/arm_irq_vector_table.c

```diff
@@ -10,7 +10,59 @@
 #include <linker/sections.h>
 
 
-struct k_sem sem[3];
+/*
+ * Offset (starting from the beginning of the vector table)
+ * of the location where the ISRs will be manually installed.
+ */
+#define _ISR_OFFSET 0
+
+#if defined(CONFIG_SOC_SERIES_NRF52X)
+/* The customized solution for nRF52X-based platforms
+ * requires that the POWER_CLOCK_IRQn line equals 0.
+ */
+BUILD_ASSERT_MSG(POWER_CLOCK_IRQn == 0,
+	"POWER_CLOCK_IRQn != 0. Consider rework manual vector table.");
+
+/* The customized solution for nRF52X-based platforms
+ * requires that the RTC1 IRQ line equals 17 and is larger
+ * than the CONFIG_NUM_IRQS.
+ */
+BUILD_ASSERT_MSG(RTC1_IRQn == 17,
+	 "RTC1_IRQn != 17. Consider rework manual vector table.");
+BUILD_ASSERT_MSG(RTC1_IRQn >= CONFIG_NUM_IRQS,
+	 "RTC1_IRQn < NUM_IRQs. Consider rework manual vector table.");
+
+#undef _ISR_OFFSET
+/* Interrupt line 0 is used by POWER_CLOCK */
+#define _ISR_OFFSET 1
+
+#elif defined(CONFIG_SOC_SERIES_NRF91X)
+/* The customized solution for nRF91X-based platforms
+ * requires that the POWER_CLOCK_IRQn line equals 5.
+ */
+BUILD_ASSERT_MSG(CLOCK_POWER_IRQn == 5,
+	"POWER_CLOCK_IRQn != 5."
+	"Consider rework manual vector table.");
+
+/* The customized solution for nRF91X-based platforms
+ * requires that the RTC1 IRQ line equals 21 and is larger
+ * than the CONFIG_NUM_IRQS.
+ */
+BUILD_ASSERT_MSG(RTC1_IRQn == 21,
+	 "RTC1_IRQn != 21. Consider rework manual vector table.");
+BUILD_ASSERT_MSG(RTC1_IRQn >= CONFIG_NUM_IRQS,
+	 "RTC1_IRQn < NUM_IRQs. Consider rework manual vector table.");
+
+#undef _ISR_OFFSET
+/* Interrupt lines 8-10 is the first set of consecutive interrupts implemented
+ * in nRF9160 SOC.
+ */
+#define _ISR_OFFSET 8
+
+#endif /* CONFIG_SOC_SERIES_NRF52X */
+
+
+struct k_sem sem[CONFIG_NUM_IRQS];
```

Comment:
The size of this array must be at least `3`, because of `sem[2]` accesses done below, for instance [here](https://github.com/zephyrproject-rtos/zephyr/pull/13903/files#diff-86ebba4ac9106b75b158163bc8c49e7bR140). Therefore, it cannot be `CONFIG_NUM_IRQS`, it should be kept at `3` as it was.

Reply:
Reverted.